Code/Editor/EditorEngineProcessFramework/IPC/ProcessCommunicationChannel.cpp

```diff
@@ -101,6 +101,12 @@ ezResult ezProcessCommunicationChannel::WaitForMessage(const ezRTTI* pMessageTyp
   while (m_pWaitForMessageType != nullptr)
   {
     m_pChannel->WaitForMessages();
+    if (!m_pChannel->IsConnected())
```

Comment:
Why was this additional check necessary?

Reply:
I had the following situation
* The editor would wait for a message on the QT main thread
* The EditorEngineProcess would crash before sending the message
* The editor now hangs in this loop forever. As this happens on the QT main thread, the editor would also be unresponsive, e.g. all of the UI is unusable.